### DISCUSSION TITLE
How to login?

### DISCUSSION THREAD

**Discussion by yggcwx** [Stirling-Tools/Stirling-PDF#1269] [Q&A] [answered]
Friends, I deployed Stirling-PDF via docker-compose, and after successful deployment, I set the parameters “DOCKER_ENABLE_SECURITY=true”, “ enableLogin: true” ‘csrfDisabled: true’ and other parameters, however, my project home page does not appear on the ‘Login’ button or the entrance, using the “http://<your_stirling_pdf_host>:<your_stirling_pdf_port>/login”-style link to log in , then prompted that there is no such page. So, how do I set up and operate the administrator page? Thank you!

如何登录？
朋友们，我是通过docker-compose方式部署的Stirling-PDF。部署成功后，我按照能搜索到的资料，分别设置了“DOCKER_ENABLE_SECURITY=true”、“enableLogin: true”“csrfDisabled: true”等参数，然而，我的项目主页上根本不出现“登录”按钮或入口，使用“http://<your_stirling_pdf_host>:<your_stirling_pdf_port>/login”样式的链接登录 ，则提示无此页面。那么，我究竟该如何设置、操作，才能登录管理员页面呢？谢谢！
![2024-05-22_16-03-06](https://github.com/Stirling-Tools/Stirling-PDF/assets/90705180/f4be2b5e-dd76-4052-bb22-a704e17b6bf9)
![2024-05-22_16-32-38](https://github.com/Stirling-Tools/Stirling-PDF/assets/90705180/2f0c1f94-63be-4291-bc6a-dd2238cd1a04)

**Comment by Frooodle** [comment]
> Friends, I deployed Stirling-PDF via docker-compose, and after successful deployment, I set the parameters “DOCKER_ENABLE_SECURITY=true”, “ enableLogin: true” ‘csrfDisabled: true’ and other parameters, however, my project home page does not appear on the ‘Login’ button or the entrance, using the “http://<your_stirling_pdf_host>:<your_stirling_pdf_port>/login”-style link to log in , then prompted that there is no such page. So, how do I set up and operate the administrator page? Thank you!
> 
> 如何登录？
> 朋友们，我是通过docker-compose方式部署的Stirling-PDF。部署成功后，我按照能搜索到的资料，分别设置了“DOCKER_ENABLE_SECURITY=true”、“enableLogin: true”“csrfDisabled: true”等参数，然而，我的项目主页上根本不出现“登录”按钮或入口，使用“http://<your_stirling_pdf_host>:<your_stirling_pdf_port>/login”样式的链接登录 ，则提示无此页面。那么，我究竟该如何设置、操作，才能登录管理员页面呢？谢谢！
> ![2024-05-22_16-03-06](https://github.com/Stirling-Tools/Stirling-PDF/assets/90705180/f4be2b5e-dd76-4052-bb22-a704e17b6bf9)
> ![2024-05-22_16-32-38](https://github.com/Stirling-Tools/Stirling-PDF/assets/90705180/2f0c1f94-63be-4291-bc6a-dd2238cd1a04)
> 

Be sure to use a versioned tag not alpha 
And if you still have issues please post docker compose and startup logs

**Reply by yggcwx** [reply]
For this deployment, I'm using the “latest” tag and not pulling a mirror of the “alpha” tag. Are you saying that pulling the “0.24.4” tag image will solve the problem and bring up the “login” portal or page?
本次部署，我使用的是“latest”标签，并没有拉取“alpha”标签的镜像啊。您的意思，是让我拉取“0.24.4”标签的镜像，就能解决问题，出现”登录“入口或页面吗？
![2024-05-22_21-25-55](https://github.com/Stirling-Tools/Stirling-PDF/assets/90705180/21d85111-1bdc-494d-afdd-02aa5d56d03c)
![2024-05-22_21-38-15](https://github.com/Stirling-Tools/Stirling-PDF/assets/90705180/d13016f9-c815-4d5f-8bd0-7726ea0e1223)
![2024-05-22_21-28-44](https://github.com/Stirling-Tools/Stirling-PDF/assets/90705180/6c4bda90-8b54-4d11-8ecc-1f3d5f425cf8)

**Reply by Frooodle** [reply]
Compose file and startup logs please :)

**Reply by yggcwx** [reply]
##### settings.yml
`# Welcome to settings file
# Remove comment marker # if on start of line to enable the configuration
# If you want to override with environment parameter follow parameter naming SECURITY_INITIALLOGIN_USERNAME

security:
  enableLogin: true # set to 'true' to enable login
  csrfDisabled: true # Set to 'true' to disable CSRF protection (not recommended for production)
  loginAttemptCount: 5 # lock user account after 5 tries
  loginResetTimeMinutes : 120 # lock account for 2 hours after x attempts
  initialLogin:
    username: "admin" # Initial username for the first login
    password: "stirling" # Initial password for the first login
  oauth2:
    enabled: true # set to 'true' to enable login (Note: enableLogin must also be 'true' for this to work)
    issuer: "" # set to any provider that supports OpenID Connect Discovery (/.well-known/openid-configuration) end-point
    clientId: "" # Client ID from your provider
    clientSecret: "" # Client Secret from your provider
    autoCreateUser: false # set to 'true' to allow auto-creation of non-existing users
    useAsUsername: "email" # Default is 'email'; custom fields can be used as the username
    scopes: "openid, profile, email" # Specify the scopes for which the application will request permissions
    provider: "google" # Set this to your OAuth provider's name, e.g., 'google' or 'keycloak'

system:
  defaultLocale: 'zh-CN' # Set the default language (e.g. 'de-DE', 'fr-FR', etc)
  googlevisibility: false # 'true' to allow Google visibility (via robots.txt), 'false' to disallow
  enableAlphaFunctionality: false # Set to enable functionality which might need more testing before it fully goes live (This feature might make no changes)
  showUpdate: false # see when a new update is available
  showUpdateOnlyAdmin: true # Only admins can see when a new update is available, depending on showUpdate it must be set to 'true'
  customHTMLFiles: false # enable to have files placed in /customFiles/templates override the existing template html files

ui:
  appName: 零维PDF # Application's visible name
  homeDescription: 一站式解决所有PDF需求。小众使用，不对外。 # Short description or tagline shown on homepage.
  appNameNavbar: 零维PDF # Name displayed on the navigation bar

endpoints:
  toRemove: [] # List endpoints to disable (e.g. ['img-to-pdf', 'remove-pages'])
  groupsToRemove: [] # List groups to disable (e.g. ['LibreOffice'])

metrics:
  enabled: true # 'true' to enable Info APIs (`/api/*`) endpoints, 'false' to disable

# Automatically Generated Settings (Do Not Edit Directly)
AutomaticallyGenerated:
  key: example
`

##### docker-compose.yml
`version: '3.3'
services:
  stirling-pdf:
    image: frooodle/s-pdf:latest
    ports:
      - '8080:8080'
    volumes:
      - /home/stirling-pdf/trainingData:/usr/share/tessdata #Required for extra OCR languages
      - /home/stirling-pdf/extraConfigs:/configs
      - /home/stirling-pdf/customFiles:/customFiles/
      - /home/stirling-pdf/logs:/logs
    environment:
      - DOCKER_ENABLE_SECURITY=true`

##### Logs
`07:59:39.833 [main] INFO  o.a.coyote.http11.Http11NioProtocol - Initializing ProtocolHandler ["http-nio-8080"]
07:59:39.838 [main] INFO  o.a.catalina.core.StandardService - Starting service [Tomcat]
07:59:39.839 [main] INFO  o.a.catalina.core.StandardEngine - Starting Servlet engine: [Apache Tomcat/10.1.19]
07:59:39.914 [main] INFO  o.a.c.c.C.[Tomcat].[localhost].[/] - Initializing Spring embedded WebApplicationContext
07:59:39.917 [main] INFO  o.s.b.w.s.c.ServletWebServerApplicationContext - Root WebApplicationContext: initialization completed in 2736 ms
07:59:40.309 [main] INFO  s.software.SPDF.SPdfApplication - Running configs ApplicationProperties [security=Security [enableLogin=true, oauth2=OAUTH2 [enabled=true, issuer=, clientId=, clientSecret=NULL, autoCreateUser=false, useAsUsername=email, providergoogle, scopes=[openid, profile, email]], initialLogin=InitialLogin [username=admin, password=MASKED],  csrfDisabled=true], system=System [defaultLocale=zh-CN, googlevisibility=false, enableAlphaFunctionality=false, showUpdate=false, showUpdateOnlyAdmin=true], ui=UserInterface [appName=零维PDF, homeDescription=一站式解决所有PDF需求。小众使用，不对外。, appNameNavbar=零维PDF], endpoints=Endpoints [toRemove=[], groupsToRemove=[]], metrics=Metrics [enabled=true], automaticallyGenerated=AutomaticallyGenerated [key=MASKED], autoPipeline=null]
07:59:40.956 [main] INFO  s.s.S.config.EndpointConfiguration - Disabling pdf-to-book
07:59:40.956 [main] INFO  s.s.S.config.EndpointConfiguration - Disabling book-to-pdf
07:59:41.724 [main] INFO  o.s.b.a.e.web.EndpointLinksResolver - Exposing 1 endpoint(s) beneath base path '/actuator'
07:59:41.790 [main] INFO  o.a.coyote.http11.Http11NioProtocol - Starting ProtocolHandler ["http-nio-8080"]
07:59:41.812 [main] INFO  o.s.b.w.e.tomcat.TomcatWebServer - Tomcat started on port 8080 (http) with context path ''
07:59:41.832 [main] INFO  s.software.SPDF.SPdfApplication - Started SPdfApplication in 5.798 seconds (process running for 7.001)
07:59:42.956 [main] INFO  s.software.SPDF.SPdfApplication - Stirling-PDF Started.
07:59:42.957 [main] INFO  s.software.SPDF.SPdfApplication - Navigate to http://localhost:8080
07:59:55.117 [http-nio-8080-exec-1] INFO  o.a.c.c.C.[Tomcat].[localhost].[/] - Initializing Spring DispatcherServlet 'dispatcherServlet'
07:59:55.117 [http-nio-8080-exec-1] INFO  o.s.web.servlet.DispatcherServlet - Initializing Servlet 'dispatcherServlet'
07:59:55.119 [http-nio-8080-exec-1] INFO  o.s.web.servlet.DispatcherServlet - Completed initialization in 2 ms
08:05:11.608 [http-nio-8080-exec-8] INFO  o.s.api.AbstractOpenApiResource - Init duration for springdoc-openapi is: 967 ms
Copying original files without overwriting existing files
Running Stirling PDF with DOCKER_ENABLE_SECURITY=false and VERSION_TAG=0.24.4
Setting permissions and ownership for necessary directories...
Picked up JAVA_TOOL_OPTIONS:  -XX:MaxRAMPercentage=75
 ____ _____ ___ ____  _     ___ _   _  ____       ____  ____  _____
/ ___|_   _|_ _|  _ \| |   |_ _| \ | |/ ___|     |  _ \|  _ \|  ___|
\___ \ | |  | || |_) | |    | ||  \| | |  _ _____| |_) | | | | |_
 ___) || |  | ||  _ <| |___ | || |\  | |_| |_____|  __/| |_| |  _|
|____/ |_| |___|_| \_\_____|___|_| \_|\____|     |_|   |____/|_|
Powered by Spring Boot 3.2.4
08:28:40.385 [main] INFO  s.software.SPDF.SPdfApplication - Starting SPdfApplication v0.24.4 using Java 21.0.3 with PID 10 (/app.jar started by stirlingpdfuser in /)
08:28:40.397 [main] INFO  s.software.SPDF.SPdfApplication - No active profile set, falling back to 1 default profile: "default"
08:28:43.173 [main] INFO  o.s.b.w.e.tomcat.TomcatWebServer - Tomcat initialized with port 8080 (http)
08:28:43.191 [main] INFO  o.a.coyote.http11.Http11NioProtocol - Initializing ProtocolHandler ["http-nio-8080"]
08:28:43.196 [main] INFO  o.a.catalina.core.StandardService - Starting service [Tomcat]
08:28:43.196 [main] INFO  o.a.catalina.core.StandardEngine - Starting Servlet engine: [Apache Tomcat/10.1.19]
08:28:43.254 [main] INFO  o.a.c.c.C.[Tomcat].[localhost].[/] - Initializing Spring embedded WebApplicationContext
08:28:43.256 [main] INFO  o.s.b.w.s.c.ServletWebServerApplicationContext - Root WebApplicationContext: initialization completed in 2718 ms
08:28:43.526 [main] INFO  s.software.SPDF.SPdfApplication - Running configs ApplicationProperties [security=Security [enableLogin=true, oauth2=OAUTH2 [enabled=true, issuer=, clientId=, clientSecret=NULL, autoCreateUser=false, useAsUsername=email, providergoogle, scopes=[openid, profile, email]], initialLogin=InitialLogin [username=admin, password=MASKED],  csrfDisabled=true], system=System [defaultLocale=zh-CN, googlevisibility=false, enableAlphaFunctionality=false, showUpdate=false, showUpdateOnlyAdmin=true], ui=UserInterface [appName=零维PDF, homeDescription=一站式解决所有PDF需求。小众使用，不对外。, appNameNavbar=零维PDF], endpoints=Endpoints [toRemove=[], groupsToRemove=[]], metrics=Metrics [enabled=true], automaticallyGenerated=AutomaticallyGenerated [key=MASKED], autoPipeline=null]
08:28:44.296 [main] INFO  s.s.S.config.EndpointConfiguration - Disabling pdf-to-book
08:28:44.296 [main] INFO  s.s.S.config.EndpointConfiguration - Disabling book-to-pdf
08:28:45.199 [main] INFO  o.s.b.a.e.web.EndpointLinksResolver - Exposing 1 endpoint(s) beneath base path '/actuator'
08:28:45.260 [main] INFO  o.a.coyote.http11.Http11NioProtocol - Starting ProtocolHandler ["http-nio-8080"]
08:28:45.321 [main] INFO  o.s.b.w.e.tomcat.TomcatWebServer - Tomcat started on port 8080 (http) with context path ''
08:28:45.359 [main] INFO  s.software.SPDF.SPdfApplication - Started SPdfApplication in 6.226 seconds (process running for 8.406)
08:28:46.506 [main] INFO  s.software.SPDF.SPdfApplication - Stirling-PDF Started.
08:28:46.506 [main] INFO  s.software.SPDF.SPdfApplication - Navigate to http://localhost:8080
08:32:06.129 [http-nio-8080-exec-2] INFO  o.a.c.c.C.[Tomcat].[localhost].[/] - Initializing Spring DispatcherServlet 'dispatcherServlet'
08:32:06.129 [http-nio-8080-exec-2] INFO  o.s.web.servlet.DispatcherServlet - Initializing Servlet 'dispatcherServlet'
08:32:06.130 [http-nio-8080-exec-2] INFO  o.s.web.servlet.DispatcherServlet - Completed initialization in 1 ms
08:33:40.337 [http-nio-8080-exec-6] INFO  o.a.coyote.http11.Http11Processor - Error parsing HTTP request header
 Note: further occurrences of HTTP request parsing errors will be logged at DEBUG level.
java.lang.IllegalArgumentException: Invalid character found in method name [0x030x000x00/*0xe00x000x000x000x000x00Cookie: ]. HTTP method names must be tokens
	at org.apache.coyote.http11.Http11InputBuffer.parseRequestLine(Http11InputBuffer.java:407)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:264)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:896)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1744)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
	at java.base/java.lang.Thread.run(Thread.java:1583)
09:10:04.998 [http-nio-8080-exec-5] INFO  o.a.coyote.http11.Http11Processor - Error parsing HTTP request header
 Note: further occurrences of HTTP request parsing errors will be logged at DEBUG level.
java.lang.IllegalArgumentException: Invalid character found in method name [0x030x000x00/*0xe00x000x000x000x000x00Cookie: ]. HTTP method names must be tokens
	at org.apache.coyote.http11.Http11InputBuffer.parseRequestLine(Http11InputBuffer.java:407)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:264)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:896)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1744)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
	at java.base/java.lang.Thread.run(Thread.java:1583)
11:08:31.548 [http-nio-8080-exec-3] INFO  o.a.coyote.http11.Http11Processor - Error parsing HTTP request header
 Note: further occurrences of HTTP request parsing errors will be logged at DEBUG level.
java.lang.IllegalArgumentException: Invalid character found in method name [0x030x000x00/*0xe00x000x000x000x000x00Cookie: ]. HTTP method names must be tokens
	at org.apache.coyote.http11.Http11InputBuffer.parseRequestLine(Http11InputBuffer.java:407)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:264)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:896)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1744)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
	at java.base/java.lang.Thread.run(Thread.java:1583)
14:01:26.667 [http-nio-8080-exec-8] INFO  o.a.coyote.http11.Http11Processor - Error parsing HTTP request header
 Note: further occurrences of HTTP request parsing errors will be logged at DEBUG level.
java.lang.IllegalArgumentException: Invalid character found in method name [0x030x000x00/*0xe00x000x000x000x000x00Cookie: ]. HTTP method names must be tokens
	at org.apache.coyote.http11.Http11InputBuffer.parseRequestLine(Http11InputBuffer.java:407)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:264)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:896)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1744)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
	at java.base/java.lang.Thread.run(Thread.java:1583)`

**Reply by yggcwx** [reply]
Have uploaded all the required information, please take the time to look at it and point me in the right direction: how can I log in to the admin page.
已上传所有需要的信息，请抽空看看，并指点我：怎么才能登录到管理员页面。

**Comment by yggcwx** [comment]
I set it up with reference to [issues1040] (https://github.com/Stirling-Tools/Stirling-PDF/issues/1040) and only once did the login box appear, but the page opened slowly. Uninstall and reinstall, the settings are still referring to [issues1040] (https://github.com/Stirling-Tools/Stirling-PDF/issues/1040), but the login box doesn't appear anyway, which is weird. Who's going to save me.

我参照[issues1040](https://github.com/Stirling-Tools/Stirling-PDF/issues/1040)进行了设置，只有一次出现过登录框，但页面打开又很慢。卸载后重新安装，设置仍然是参照[issues1040](https://github.com/Stirling-Tools/Stirling-PDF/issues/1040)，但无论如何都不出现登录框了，真是奇怪。谁来救救我。

**Reply by yggcwx** [reply, marked as answer]
很神奇，换了台主机，我的问题就解决了。

**Reply by xingzhej** [reply]
> 很神奇，换了台主机，我的问题就解决了。

请用英文，这里是English Tools。PDF转word的时候有很多文本框，怎么处理的兄弟。
